lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+
+if (!inspector || !inspector.asyncTaskScheduled) {
+  exports.setup = function() {};
+  return;
+}
+
+const hook = createHook({
+  init(asyncId, type, triggerId, resource) {
+    // It's difficult to tell which tasks will be recurring and which won't,
+    // therefore we mark all tasks as recurring. Based on the discussion
+    // in https://github.com/nodejs/node/pull/13870#discussion_r124515293,
+    // this should be fine as long as we call asyncTaskCanceled() too.
+    const recurring = true;
+    inspector.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    inspector.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    inspector.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    inspector.asyncTaskCanceled(asyncId);
+  },
+});
+
+exports.setup = function() {
+  inspector.registerAsyncHook(() => hook.enable(), () => hook.disable());
```

Comment:
Right now, we never use the `disable` callback, because AFAICT, the inspector mode is never disabled after it has been enabled. Am I missing something? Should I remove `disable` callback from the C++ layer to simplify things in the spirit of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)?